core/commands/swarm.go

```diff
@@ -557,10 +560,43 @@ func parseAddresses(addrs []string) (iaddrs []iaddr.IPFSAddr, err error) {
 	return
 }
 
+// parseMultiaddrs is a function that takes in a slice of peer multiaddr
+// and returns slices of multiaddrs and peerids
+func parseMultiaddrs(maddrs []ma.Multiaddr) (iaddrs []iaddr.IPFSAddr, err error) {
+	iaddrs = make([]iaddr.IPFSAddr, len(maddrs))
+	for i, maddr := range maddrs {
+		iaddrs[i], err = iaddr.ParseMultiaddr(maddr)
+		if err != nil {
+			return nil, cmds.ClientError("invalid peer address: " + err.Error())
+		}
+	}
+	return
+}
+
 // peersWithAddresses is a function that takes in a slice of string peer addresses
 // (multiaddr + peerid) and returns a slice of properly constructed peers
 func peersWithAddresses(addrs []string) ([]pstore.PeerInfo, error) {
-	iaddrs, err := parseAddresses(addrs)
+	// resolve the addrs
+
+	var maddrs []ma.Multiaddr
+	for _, addr := range addrs {
+		maddr, err := ma.NewMultiaddr(addr)
+		if err != nil {
+			return nil, err
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
I've looked through this whole PR and everything makes sense to me :+1:. I do have a question about how these timeouts are specified in go-ipfs (both here and more generally). I notice lines like the following (and there are a handful):

https://github.com/ipfs/go-ipfs/blob/master/core/bootstrap.go#L60

But also lines like yours (e.g.):

https://github.com/ipfs/go-ipfs/blob/master/core/commands/pubsub.go#L178

Is there a rule of thumb for deciding when to specify these timeouts with a name vs inline?

Reply:
Well, we should probably name this. We just get lazy sometimes.